src/mappings/mappingHandlers.ts

```diff
@@ -284,3 +286,83 @@ export async function handleDelegatorWithdrawRewardEvent(event: CosmosEvent): Pr
   claim.denom = denom;
   await claim.save();
 }
+
+export async function handleNativeBalanceDecrement(event: CosmosEvent): Promise<void> {
+  logger.info(`[handleNativeBalanceDecrement] (tx ${event.tx.hash}): indexing event ${event.idx + 1} / ${event.tx.tx.events.length}`)
+  logger.debug(`[handleNativeBalanceDecrement] (event.event): ${JSON.stringify(event.event, null, 2)}`)
+  logger.debug(`[handleNativeBalanceDecrement] (event.log): ${JSON.stringify(event.log, null, 2)}`)
+
+  // sample event.event.attributes:
+  // [
+  //   {"key":"spender","value":"fetch1jv65s3grqf6v6jl3dp4t6c9t9rk99cd85zdctg"},
+  //   {"key":"amount","value":"75462013217046121atestfet"},
+  //   {"key":"spender","value":"fetch1wurz7uwmvchhc8x0yztc7220hxs9jxdjdsrqmn"},
+  //   {"key":"amount","value":"100atestfet"}
+  // ]
+  let spendEvents = [];
+  event.event.attributes.map((e, i) => {
+    if (e.key !== "spender") {
+      return
+    }
+    const spender = e.value;
+    const amountStr = event.event.attributes[i+1].value;
+
+    const coin = parseCoins(amountStr)[0];
+    const amount = BigInt(0) - BigInt(coin.amount); // save a negative amount for a "spend" event
+    spendEvents.push({spender: spender, amount: amount, denom: coin.denom})
+  });
+  
+  for (const i in spendEvents) {
+    const spendEvent = spendEvents[i];
+    await saveNativeBalanceEvent(`${messageId(event)}-spend-${i}`, spendEvent.spender, spendEvent.amount, spendEvent.denom, event);
+  }
+}
+
+export async function handleNativeBalanceIncrement(event: CosmosEvent): Promise<void> {
+  logger.info(`[handleNativeBalanceIncrement] (tx ${event.tx.hash}): indexing event ${event.idx + 1} / ${event.tx.tx.events.length}`)
+  logger.debug(`[handleNativeBalanceIncrement] (event.event): ${JSON.stringify(event.event, null, 2)}`)
+  logger.debug(`[handleNativeBalanceIncrement] (event.log): ${JSON.stringify(event.log, null, 2)}`)
+
+  // sample event.event.attributes:
+  // [
+  //   {"key":"receiver","value":"fetch1jv65s3grqf6v6jl3dp4t6c9t9rk99cd85zdctg"},
+  //   {"key":"amount","value":"75462013217046121atestfet"},
+  //   {"key":"receiver","value":"fetch1wurz7uwmvchhc8x0yztc7220hxs9jxdjdsrqmn"},
+  //   {"key":"amount","value":"100atestfet"}
+  // ]
+  let receiveEvents = [];
+  event.event.attributes.map((e, i) => {
```

Comment:
nit: not a big deal, but since we're not returning anything here we could use a for over [`Object.entries`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries) loop instead which I think is more idomatic.

```suggestion
  for (const [e, i] of Object.entries(event.event.attributes)) {
```